noxfile.py

```diff
@@ -0,0 +1,34 @@
+import nox
+from nox_poetry import session
+
+
+PYTHON_VERSIONS = ["3.6", "3.7", "3.8", "3.9"]
+SPHINX_VERSIONS = ["2.2", "2.3", "2.4", "3.0", "3.2"]
+TEST_DEPENDENCIES = [
```

Comment:
so this isn't defining the dependency versions, this is defining the test matrix (of course the test matrix fits *within* the version specification)

Reply:
ah i remember why. it's because the test dependencies are a subset of the `dev-dependencies`. it's not necessary to install all of the dev dependencies. this creates a significant time saving if running the full test matrix sequentially.

the difference is even more striking when you're doing something like linting, for which only 1 or 2 of the dev-dependencies is required.

a neater solution would be to group the dev dependencies (keep an eye on https://github.com/python-poetry/poetry/issues/1644)